File: src/python/handlers/wset.py
from utils import wind_data_to_dict, handle_error
from WindPy import w


def handle_wset(params: dict) -> dict:
    table_name = params["tableName"]
    options = params.get("options", "")

    data = w.wset(table_name, options)

    if data.ErrorCode != 0:
        raise RuntimeError(handle_error(data.ErrorCode))

    return wind_data_to_dict(data)

File: src/tools/wset.ts
import { z } from "zod";
import { runBridge } from "../bridge/runner.js";

export const wsetTool = {
  name: "wind_wset",
  description: "Query Wind dataset tables (e.g. index constituents, sector members) using Wind w.wset().",
  inputSchema: z.object({
    tableName: z.string().describe("Wind dataset table name, e.g. 'IndexConstituent'"),
    options: z.string().optional().describe("Wind options string, e.g. 'date=2024-01-01;windcode=000300.SH'"),
  }),
  handler: async (params: Record<string, unknown>) => {
    const result = await runBridge({ function: "wset", params });
    if (!result.ok) throw new Error(result.error);
    return result.data;
  },
};
